### PR TITLE
Updated to beta web3

### PR DESCRIPTION
**Pull request by linusnorton**
This PR updates TypeChain to support the web3 1.0 API. 

This code now returns strings in place of BigNumber's to keep compatibility with web3's API.

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -7,13 +7,14 @@ yarn test:unit
 
 echo "Praparing for integration tests"
 
-ABI_DIR="./test/integration/abis"
+cd ./test/integration/contracts/
+ABI_DIR="../abis"
 rm -rf $ABI_DIR
 
 echo "Generating ABIs for sample contracts"
-solc --abi ./test/integration/contracts/* --bin -o $ABI_DIR
```
Comment (linusnorton) [inline scripts/test.sh]: Added a dev dependency for solc so it now uses the inbuilt solcjs

File: test/integration/ContractWithOverloads.spec.ts [hunk 2]
```diff
@@ -2,21 +2,24 @@ import { expect } from "chai";
 import { deployContract } from "./utils/web3Contracts";
 
 import { web3 } from "./web3";
-import { ContractWithOverloads } from "./abis/ContractWithOverloads";
+import { __ContractWithOverloads_sol_ContractWithOverloads as ContractWithOverloads } from "./abis/__ContractWithOverloads_sol_ContractWithOverloads";
```
Comment (linusnorton) [inline test/integration/ContractWithOverloads.spec.ts]: This was a side effect of switching to solcjs

File: test/integration/ContractWithOverloads.spec.ts [hunk 3]
```diff
@@ -2,21 +2,24 @@ import { expect } from "chai";
...
   it("should be able to access counter", async () => {
     const contractWithOverloads = await ContractWithOverloads.createAndValidate(
       web3,
       contractAddress
     );
-    expect((await contractWithOverloads.counter).toNumber()).to.be.eq(0);
+    const result = await contractWithOverloads.counter;
+
+    // numbers now returned as string
+    expect(result.toNumber()).to.be.eq(0);
```
Comment (linusnorton) [inline test/integration/ContractWithOverloads.spec.ts]: Here be problems with BigNumber

File: test/integration/Formatting.spec.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ describe("Formatting of generated files", () => {
 
     files.forEach(filePath => {
       expect(
-        check(readFileSync(join(__dirname, "./abis/ContractWithOverloads.ts"), "utf-8"), config!),
```
Comment (linusnorton) [inline test/integration/Formatting.spec.ts]: Not sure this line was doing what you intended :)

File: tsconfig.json [hunk 5]
```diff
@@ -23,7 +23,6 @@
     "./lib/**/*.ts",
     "./test/**/*.ts",
     "./typings/*.ts",
-    "./node_modules/web3-typescript-typings/index.d.ts",
```
Comment (linusnorton) [inline tsconfig.json]: No longer necessary

**Comment by linusnorton** [comment]
@krzkaczor I've merged next and retargeted. I've left the version as 1.0.0 but it's up to you if you want to go down that route.

File: tslint.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "typestrict",
   "rules": {
-    "no-console": true,
+    "no-console": false,
```
Comment (krzkaczor) [inline tslint.json]: Could you revert this change? This is a useful rule, it's just typechain should encapsulate logging in one place but now it's everywhere :/

File: test/integration/web3.ts [hunk 7]
```diff
@@ -1,16 +1,15 @@
 import * as ganache from "ganache-cli";
-
-import * as Web3 from "web3";
-import { promisify } from "bluebird";
+import Web3 from "web3";
```
Comment (krzkaczor) [inline test/integration/web3.ts]: do `import * as Web3 from "web3";`  here

File: test/integration/web3.ts [hunk 8]
```diff
@@ -1,16 +1,15 @@
...
 
 export const GAS_LIMIT_STANDARD = 1000000;
 
 export let web3: Web3;
 export let accounts: string[];
 
 export async function createNewBlockchain() {
-  const web3 = new Web3(ganache.provider());
-  const accounts = await promisify(web3.eth.getAccounts, { context: web3.eth })();
+  const Derp = require("web3");
```
Comment (krzkaczor) [inline test/integration/web3.ts]: after changing import this should not be necessary.

**Comment by krzkaczor** [comment]
@linusnorton thanks! I will figure out something regarding the version.

**Comment by jpeletier** [comment]
Hello all. Very interesting project. Any news on this PR? I don't mind making the requested changes and creating a new PR out of them. Please let me know.

**Comment by krzkaczor** [comment]
@jpeletier Hey! I am willing to merge it of course! Someone has to just fix comments. It would be cool if you could work on this @jpeletier
